web/src/bootstrap_typeahead.ts

```diff
@@ -373,7 +375,29 @@ export class Typeahead<ItemType extends string | object> {
                 // We expect the typeahead creator to handle when to hide / show the typeahead.
                 trigger: "manual",
                 arrow: false,
-                offset: [0, 2],
+                offset({placement, reference}) {
+                    if (input_element.type === "textarea" && input_element.$element[0]) {
+                        const caret = getCaretCoordinates(
+                            input_element.$element[0],
+                            input_element.$element[0].selectionStart,
+                        );
+                        // Used to consider the scroll height of textbox in the vertical offset.
+                        const scrollTop = input_element.$element.scrollTop() ?? 0;
+                        if (placement === "top-start") {
+                            // Gap separates the typeahead and caret by 4px.
+                            const gap = 4;
+                            return [caret.left, -caret.top + scrollTop + gap];
+                        }
+                        // In bottom-start, the offset is calculated from bottom of the popper reference.
+                        if (placement === "bottom-start") {
+                            // Height of the reference is the input_element height.
+                            const field_height = reference.height;
+                            const distance = field_height - caret.top + scrollTop - caret.height;
+                            return [caret.left, -distance];
+                        }
+                    }
+                    return [0, 0];
```

Comment:
Why did this change from `[0, 2]` to `[0, 0]`? I would think that we'd be hoping to not change anything at all in the case that it's not a textarea typeahead.

Reply:
Must be a mistake while resolving previous conflicts. Fixed it!

Reply:
Indeed, that was introduced in 52ac602acb2b280490261e85ed908d3ff5d423ba. Resolving merge conflicts correctly is worth spending serious time on -- reverting someone else's changes by accident usually feels pretty bad!